packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -732,9 +735,20 @@ export async function createRouteManifest(
 		}
 	}
 
-	if (!dev) {
-		await runHookRoutesResolved({ routes, settings, logger });
+	if (dev) {
+		ensure404Route({ routes });
+		injectImageEndpoint(settings, { routes }, 'dev');
+		injectServerIslandRoute(settings.config, { routes });
+	} else if (settings.buildOutput === 'server') {
+		injectImageEndpoint(settings, { routes }, 'build');
+		// We can't only inject this route when server islands are used because:
+		// - To know if we use serve islands, we need to build
+		// - The build runs before we can inject the server island route
+		injectServerIslandRoute(settings.config, { routes });
+		// In SSR, a 404 route is injected in the App directly for some special handling,
+		// it must not appear in the manifest
 	}
+	await runHookRoutesResolved({ routes, settings, logger });
```

Comment:
Is this intended? Now, the hook is called even in dev.

Reply:
Yes this is intended, it was called in dev before but through a different place. It's united now